backend/globaleaks/handlers/auth/__init__.py

```diff
@@ -97,8 +115,14 @@ def login_whistleblower(session, tid, receipt, client_using_tor, operator_id=Non
     return session
 
 
+def check_user_fiscal_code(user, fiscal_code, tid):
+    if tid == 1:
+        return True
+    if user.fiscal_code and user.fiscal_code != fiscal_code:
+        raise errors.ForbiddenOperation
+
 @transact
-def login(session, tid, username, password, authcode, client_using_tor, client_ip):
+def login(session, tid, username, password, authcode, client_using_tor, client_ip, fiscal_code):
```

Comment:
We require changes from "fiscal_code" to idp_user_id that is the international abstraction we could use to integrate an IDP. Not every country have a "fiscal code" while any idp has an idp-user-id